File: Dividend Finder - Python/dividend_finder.py
# /usr/bin/env python3
# -*- coding: utf-8 -*-
"""
@author: Matthew Thornton

This program takes an input and lists every possible whole number
multiplication that equals the specified number.
"""

class Dividend_Finder(object):
    def __init__(self):
        self.divisor = 1
    def main(self, dividend):
        divisor = self.divisor
        if dividend == 1:
            print ('1 x 1 = 1')
            return
        while divisor <= dividend:
            if dividend % divisor == 0:
                multiplyer = dividend//divisor
                print (divisor,'x', multiplyer ,'=', dividend)
                if divisor == multiplyer:
                    break
            divisor += 1
            if divisor == multiplyer:
                break

if __name__ == '__main__':
    finder = Dividend_Finder()
    while True:
        dividend = input("Please enter a number: ")
        dividend = int(dividend)
        finder.main(dividend)
        if 'y' not in input("Would you like to run again? Y/N: ").lower():
            break

File: Dividend Finder - Java/src/Dividend_Finder.java
import java.util.Scanner;

public class Dividend_Finder {

	public static void main(String[] args){
		Scanner input = new Scanner(System.in);
		while (true) {
			finder();
			System.out.print("Would you like to run again? Y/N: ");
			char choice = input.next().charAt(0);
			if (choice == 'N' || choice == 'n') {
				input.close();
				break;
			}
		}
	}

	public static void finder() {
		int divisor = 1;
		int multiplyer;
		int multiplyer_l = 1;
		Scanner input = new Scanner(System.in);
		System.out.print("Enter the Dividend or 'e' to cancel: ");
		if (input.hasNextInt()) {
			int dividend = input.nextInt();
			while (divisor <= dividend){

				multiplyer = dividend/divisor;


				if (dividend%divisor == 0){
					
					if (divisor == multiplyer_l && multiplyer_l != 1) {
						break;
					}
					multiplyer_l = multiplyer;
					System.out.println(divisor + " x " + multiplyer + " = " + dividend);
				}
				if (divisor == multiplyer) {
					break;
				}
				divisor++;
			}		
		}
		else {
			char choice = input.next().charAt(0);
			if (choice == 'e'){
				return;
			}
			else{
			System.out.println("Invalid input\n");
			}
			return;
		}
		
		
	}
}
